File: Keeper/Models/VaultKeep.cs
using System;

namespace Keeper.Models
{
    public class Vaultkeep
    {
        public int Id { get; set; }
        public string CreatorId { get; set; }
        public int VaultId { get; set; }
        public int KeepId { get; set;}
        public Profile Creator { get; set;}
        public Keep Keep { get; set; }
        public Vault vault { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: Keeper.client/src/Models/Vaultkeep.js
export class Vaultkeep{
    constructor(vkData){
        this.id = vkData.id
        this.creatorId = vkData.creatorId
        this.vaultId = vkData.vaultId
        this.keepId = vkData.keepId
        this.creator = vkData.creator || {}
        this.keep = vkData.keep || {}
        this.vault = vkData.vault || {}
        this.createdAt = vkData.createdAt
        this.updatedAt = vkData.updatedAt
    }
}